databricks/koalas/series.py

```diff
@@ -4793,19 +4793,56 @@ def dot(self, other) -> Union[Scalar, "Series"]:
 
         >>> s @ s
         14
+
+        >>> kdf = ks.DataFrame({'x': [0, 1, 2, 3], 'y': [0, -1, -2, -3]})
+        >>> kdf
+           x  y
+        0  0  0
+        1  1 -1
+        2  2 -2
+        3  3 -3
+
+        >>> with ks.option_context("compute.ops_on_diff_frames", True):
+        ...     s.dot(kdf)
+        ...
+        x    14
+        y   -14
+        dtype: int64
         """
         if isinstance(other, DataFrame):
-            raise ValueError(
-                "Series.dot() is currently not supported with DataFrame since "
-                "it will cause expansive calculation as many as the number "
-                "of columns of DataFrame"
-            )
-        if self._kdf is not other._kdf:
-            if len(self.index) != len(other.index):
+            if not self.index.equals(other.index):
                 raise ValueError("matrices are not aligned")
-        if isinstance(other, Series):
+
+            combined = combine_frames(self._kdf, other)
+            sdf = combined._internal.spark_frame
+
+            this_data_spark_column_name = combined["this"]._internal.data_spark_column_names[0]
+            that_data_spark_column_names = combined["that"]._internal.data_spark_column_names
+
+            sdf = sdf.select(
+                [
+                    (sdf[this_data_spark_column_name] * sdf[scol]).alias(scol)
```

Comment:
`scol_for(sdf, this_data_spark_column_name)`?

Reply:
Could you use `scol_for` for the other places?

Reply:
Certainly, modified!